heron/packing/src/java/com/twitter/heron/packing/roundrobin/RoundRobinPacking.java

```diff
@@ -232,10 +242,11 @@ private ByteAmount getContainerRamPadding(List<TopologyAPI.Config.KeyValue> topo
    *
    * @return containerId -&gt; list of InstanceId belonging to this container
    */
-  private Map<Integer, List<InstanceId>> getRoundRobinAllocation() {
+  private Map<Integer, List<InstanceId>> getRoundRobinAllocation(
+      int numContainer, Map<String, Integer> parallelismMap) {
     Map<Integer, List<InstanceId>> allocation = new HashMap<>();
-    int numContainer = TopologyUtils.getNumContainers(topology);
-    int totalInstance = TopologyUtils.getTotalInstance(topology);
+    int totalInstance =
```

Comment:
TopologyUtils.getTotalInstance() doesn't work? Maybe worth adding this code to TopologyUtils as a util function?

Reply:
The implementation of `TopologyUtils.getTotalInstance()`, https://github.com/twitter/heron/blob/57310fe58a808ee9cf88b91121d9d8822698bee1/heron/api/src/java/com/twitter/heron/api/utils/TopologyUtils.java#L130, seems same?

Reply:
getTotalInstance() needs a TopologyAPI.Topology which is different from Map<String, Integer>
or we may refactor getTotalInstance()? what do you think

Reply:
I think refactoring might be better. Easier to read and easier to test.

Reply:
refactored